model/src/main/java/org/projectnessie/model/UriUtil.java

```diff
@@ -44,6 +59,18 @@ private UriUtil() {}
    * @return String encoded for path use.
    */
   public static String toPathString(List<String> elements) {
-    return elements.stream().map(x -> x.replace('.', '\u0000')).collect(Collectors.joining("."));
+    return elements.stream()
+        .map(x -> x.replace(DOT, ZERO_BYTE))
+        .map(
+            x -> {
+              try {
+                return URLEncoder.encode(x, "UTF-8");
```

Comment:
Use `StandardCharsets.UTF_8`, perhaps? Then we won't have to catch `UnsupportedEncodingException`

Reply:
we're still using JDK 8 unfortunately due to https://github.com/projectnessie/nessie/blob/b750fad777b0676622185f8acc5e3f040d9aaae9/pom.xml#L158

Reply:
`StandardCharsets` exists since Java 1.7